### PR TITLE
Fixes #44: Launch donkeycar.service at startup

### LINKED ISSUE #44: Start manage.py drive at startup
Start manage.py drive at startup

### PR DESCRIPTION
**Pull request by yannis-mlgrn**
Fixes : #44

### CODE REVIEW

**Comment by Benvii** [comment]
Error [found by testing](https://github.com/mdl29/donkeycarLPH/pull/39#issuecomment-1103036752) #39 that may be also present here and should be fixed for the service to work well.

## python manage.py drive --js issue 

I noticed 2 errors : 
- Scripts are writing as root (in mycar folder)
- A missing dependency, I have attached a pip freeze ( [pipFreezeDs4Drv.txt](https://github.com/mdl29/donkeycarLPH/files/8516072/pipFreezeDs4Drv.txt) ) for further investigation we should compare it to dababycar

```
venv) pi@bluecar:~/mycar $ python manage.py drive --js
python: can't open file 'manage.py': [Errno 13] Permission denied
(venv) pi@bluecar:~/mycar $ ^Cthon manage.py drive --js
(venv) pi@bluecar:~/mycar $ sudo chown -R .
chown: missing operand after ‘.’
Try 'chown --help' for more information.
(venv) pi@bluecar:~/mycar $ ls -lisha
total 124K
254250 4.0K drwxr-xr-x 5 root root 4.0K Apr 19  2022 .
   825 4.0K drwxr-xr-x 6 pi   pi   4.0K Dec  2 02:36 ..
254257 8.0K -rwx------ 1 root root 4.8K Apr 19  2022 calibrate.py
254255  32K -rw-r--r-- 1 root root  32K Apr 19  2022 config.py
254252 4.0K drwxr-xr-x 2 root root 4.0K Apr 19  2022 data
254253 4.0K drwxr-xr-x 2 root root 4.0K Apr 19  2022 logs
254254  40K -rwx------ 1 root root  37K Apr 19  2022 manage.py
254251 4.0K drwxr-xr-x 2 root root 4.0K Apr 19  2022 models
 14107  20K -rw-r--r-- 1 root root  17K Apr 19  2022 myconfig.py
254256 4.0K -rwx------ 1 root root  728 Apr 19  2022 train.py
(venv) pi@bluecar:~/mycar $ sudo chown -R pi:pi .
(venv) pi@bluecar:~/mycar $ python manage.py drive --js
________             ______                   _________              
___  __ \_______________  /___________  __    __  ____/_____ ________
__  / / /  __ \_  __ \_  //_/  _ \_  / / /    _  /    _  __ `/_  ___/
_  /_/ // /_/ /  / / /  ,<  /  __/  /_/ /     / /___  / /_/ /_  /    
/_____/ \____//_/ /_//_/|_| \___/_\__, /      \____/  \__,_/ /_/     
                                 /____/                              

using donkey v4.3.10 ...
WARNING:donkeycar.parts.pins:pigpio was not imported.
Traceback (most recent call last):
  File "manage.py", line 27, in <module>
    from donkeycar.parts.datastore import TubHandler
  File "/home/pi/projects/donkeycar/parts/datastore.py", line 19, in <module>
    import pandas as pd
ModuleNotFoundError: No module named 'pandas'
```

**Comment by yannis-mlgrn** [comment]
> And add the following line to /etc/modules if it doesn't exists :
`i2c-dev`
How ??
I've don't fixed that

**Comment by yannis-mlgrn** [comment]
> Configuration doesn't use pigpio
Because myconfig.py is replaced by a template taken on internet. I didn't have access to a car when I have wrote this script. 
You can give me the correct script ?

File: ansible/roles/donkeycar/tasks/main.yml [hunk 1]
```diff
@@ -41,19 +36,53 @@
...
+    dest: /home/pi/.bashrc
+    line: "source ~/venv/bin/activate"
+
 - name: run setup.py
+  become: yes
+  become_user: pi
   pip:
     chdir: /home/pi/projects/
     name: .[pi]
+    become_user: pi
```
Comment (Benvii) [inline ansible/roles/donkeycar/tasks/main.yml:57]: become_user doesn't exist as a pip option

File: ansible/roles/donkeycar/tasks/main.yml [hunk 2]
```diff
@@ -41,19 +36,53 @@
...
     virtualenv: /home/pi/venv/
     extra_args: '-e'
 
-- name: Add virtualenv activation in bashrc file for pi user
```
Comment (Benvii) [inline ansible/roles/donkeycar/tasks/main.yml:56]: Why did you remove that task ? When we log into the raspberrpi using SSH it's good have the virtualenv activated.

**Review by Benvii — CHANGES_REQUESTED** [review]
Small changes needed, see review comments

**Comment by Benvii** [comment]
Failed  : 

```
    arm-image.raspbian:
    arm-image.raspbian: TASK [donkeycar : Clone the donkeycar repository and checkout in main branch] ***
    arm-image.raspbian: changed: [127.0.0.1]
    arm-image.raspbian:
    arm-image.raspbian: TASK [donkeycar : Create python3 virtual environment & install tensorflow] *****
    arm-image.raspbian: changed: [127.0.0.1]
    arm-image.raspbian:
    arm-image.raspbian: TASK [donkeycar : Add virtualenv activation in bashrc file for pi user] ********
    arm-image.raspbian: changed: [127.0.0.1]
    arm-image.raspbian:
    arm-image.raspbian: TASK [donkeycar : run setup.py] ************************************************
    arm-image.raspbian: fatal: [127.0.0.1]: FAILED! => {"changed": false, "cmd": ["/home/pi/venv/bin/pip3", "install", "-e", ".[pi]"], "msg": "stdout: Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple\nObtaining file:///home/pi/projects\n    Complete output from command python setup.py egg_info:\n    extra_files ['templates/*']\n    running egg_info\n    creating donkeycar.egg-info\n    /home/pi/venv/lib/python3.7/site-packages/setuptools/dist.py:760: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead\n      % (opt, underscore_opt)\n    error: could not create 'donkeycar.egg-info': Permission denied\n    \n    ----------------------------------------\n\n:stderr: Command \"python setup.py egg_info\" failed with error code 1 in /home/pi/projects/\n"}
    arm-image.raspbian:
    arm-image.raspbian: PLAY RECAP *********************************************************************
    arm-image.raspbian: 127.0.0.1                  : ok=14   changed=13   unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
    arm-image.raspbian:
==> arm-image.raspbian: fuser -k /tmp/armimg-227543047
Build 'arm-image.raspbian' errored after 39 minutes 28 seconds: Error executing Ansible: Non-zero exit status: 2

==> Wait completed after 39 minutes 28 seconds

==> Some builds didn't complete successfully and had errors:
--> arm-image.raspbian: Error executing Ansible: Non-zero exit status: 2
```

File: ansible/roles/donkeycar/tasks/main.yml (hunk 2 above)
Comment (yannis-mlgrn) [reply ansible/roles/donkeycar/tasks/main.yml:56]: Oups, idk why i have did that

File: ansible/roles/donkeycar/tasks/main.yml (hunk 1 above)
Comment (yannis-mlgrn) [reply ansible/roles/donkeycar/tasks/main.yml]: Also, sorry

**Comment by Benvii** [comment]
Getting this issue now : 
```
    arm-image.raspbian: TASK [donkeycar : Create donkeycar service for launch manage.py start at startup] ***
==> arm-image.raspbian: ERROR! The requested handler 'Start donckeycar.service service' was not found in either the main handlers list nor in the listening handlers list
==> arm-image.raspbian: fuser -k /tmp/armimg-673704545
Build 'arm-image.raspbian' errored after 46 minutes 57 seconds: Error executing Ansible: Non-zero exit status: 1

==> Wait completed after 46 minutes 57 seconds

==> Some builds didn't complete successfully and had errors:
--> arm-image.raspbian: Error executing Ansible: Non-zero exit status: 1

==> Builds finished but no artifacts were created.
```

**Comment by yannis-mlgrn** [comment]
> Getting this issue now :
> 
> ```
>     arm-image.raspbian: TASK [donkeycar : Create donkeycar service for launch manage.py start at startup] ***
> ==> arm-image.raspbian: ERROR! The requested handler 'Start donckeycar.service service' was not found in either the main handlers list nor in the listening handlers list
> ==> arm-image.raspbian: fuser -k /tmp/armimg-673704545
> Build 'arm-image.raspbian' errored after 46 minutes 57 seconds: Error executing Ansible: Non-zero exit status: 1
> 
> ==> Wait completed after 46 minutes 57 seconds
> 
> ==> Some builds didn't complete successfully and had errors:
> --> arm-image.raspbian: Error executing Ansible: Non-zero exit status: 1
> 
> ==> Builds finished but no artifacts were created.
> ```

Fixes : [0b43939](https://github.com/mdl29/donkeycarLPH/pull/46/commits/0b43939c966b71241b843dbdd19827c53ad6ea23)

**Comment by Benvii** [comment]
No camera at startup in the drive web interface.

**Comment by yannis-mlgrn** [comment]
> No camera at startup in the drive web interface.

It's done now? the bluecar run with camera the last time

**Comment by yannis-mlgrn** [comment]
I have a new problem : 
```________             ______                   _________              
___  __ \_______________  /___________  __    __  ____/_____ ________
__  / / /  __ \_  __ \_  //_/  _ \_  / / /    _  /    _  __ `/_  ___/
_  /_/ // /_/ /  / / /  ,<  /  __/  /_/ /     / /___  / /_/ /_  /    
/_____/ \____//_/ /_//_/|_| \___/_\__, /      \____/  \__,_/ /_/     
                                 /____/                              

using donkey v4.3.13 ...
INFO:donkeycar.config:loading config file: /home/pi/mycar/config.py
INFO:donkeycar.config:loading personal config over-rides from myconfig.py
INFO:__main__:PID: 948
cfg.CAMERA_TYPE PICAM
INFO:__main__:cfg.CAMERA_TYPE PICAM
mmal: mmal_vc_port_enable: failed to enable port vc.null_sink:in:0(OPQV): ENOSPC
mmal: mmal_port_enable: failed to enable connected port (vc.null_sink:in:0(OPQV))0x1d43240 (ENOSPC)
mmal: mmal_connection_enable: output port couldn't be enabled
Traceback (most recent call last):
  File "manage.py", line 853, in <module>
    meta=args['--meta'])
  File "manage.py", line 142, in drive
    cam = PiCamera(image_w=cfg.IMAGE_W, image_h=cfg.IMAGE_H, image_d=cfg.IMAGE_DEPTH, framerate=cfg.CAMERA_FRAMERATE, vflip=cfg.CAMERA_VFLIP, hflip=cfg.CAMERA_HFLIP)
  File "/home/pi/projects/donkeycar/parts/camera.py", line 27, in __init__
    self.camera = PiCamera() #PiCamera gets resolution (height, width)
  File "/home/pi/venv/lib/python3.7/site-packages/picamera/camera.py", line 433, in __init__
    self._init_preview()
  File "/home/pi/venv/lib/python3.7/site-packages/picamera/camera.py", line 513, in _init_preview
    self, self._camera.outputs[self.CAMERA_PREVIEW_PORT])
  File "/home/pi/venv/lib/python3.7/site-packages/picamera/renderers.py", line 558, in __init__
    self.renderer.inputs[0].connect(source).enable()
  File "/home/pi/venv/lib/python3.7/site-packages/picamera/mmalobj.py", line 2212, in enable
    prefix="Failed to enable connection")
  File "/home/pi/venv/lib/python3.7/site-packages/picamera/exc.py", line 184, in mmal_check
    raise PiCameraMMALError(status, prefix)
picamera.exc.PiCameraMMALError: Failed to enable connection: Out of resources
```

**Review by Benvii — APPROVED** [review]
Tested on monkecar, works well as expected.
